ui/src/main/webapp/src/app/services/keycloak.service.ts

```diff
@@ -114,26 +116,31 @@ export class KeycloakService {
 
     getToken(): Observable<string> {
         let promise: Promise<string> = new Promise<string>((resolve, reject) => {
-            if (!this.isLoggedIn()) {
-                reject('User is not authenticated, token is not set');
-            } else {
-                this.auth.authz.updateToken(5).success(() => {
-                    let token = <string>this.auth.authz.token;
-                    resolve(token);
-                })
-                .error(function() {
-                    reject('Failed to refresh token');
-                });
-            }
+            this.isLoggedIn().subscribe(isLoggedIn => {
+                    if (isLoggedIn) {
+                        if (this.keyCloak.isTokenExpired(KeycloakService.TOKEN_MIN_VALIDITY_SECONDS)) {
```

Comment:
This line isn't really neccessary. updateToken will still provide the old token if it isn't expired.